### PR TITLE
Milestone Delivery: Zaturn

### PR DESCRIPTION
**Pull request by pascuin**
# Milestone Delivery Checklist

- [x] The [milestone-delivery-template.md](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/deliveries/milestone-delivery-template.md) has been copied and updated.
- [x] The [invoice form :pencil:](https://forms.gle/LSRr7PCjBpEbKGh89) has been filled out for this milestone.
- [x] This pull request is being made by the same account as the accepted application.
- [x] In case of acceptance, the payment will be transferred to the BTC/ETH payment address provided in the application.
- [x] The delivery is according to the [Guidelines for Milestone Deliverables](https://github.com/w3f/Grants-Program/blob/master/docs/milestone-deliverables-guidelines.md).

Link to the application pull request: application was non-public

### CODE REVIEW

**Comment by takahser** [comment]
@pascuin thanks for submitting your delivery and sorry for the delay here. I'll take a look soon.

**Review by takahser — COMMENTED** [review]
@pascuin I didn't have time to finish my evaluation yet, but I already took a first look and want to share my findings:

- I like that you included some examples into your [docs](https://docs.acurast.com/developers/introduction/). However, I identified a bug: When clicking the _Edit Code_ button and consecutively clicking the _Show Example_  button, the site crashes: 
    
    image

- I noticed that the [acurast-processor repo](https://github.com/Acurast/acurast-processor) is currently not accessible:

    image

- since the docs don't contain step-by-step instructions or a tutorial to guide on how to the the deliveries, I'd assume that the acurast-processor repo contains theses for all 3 milestones, is that correct? If not, could you point out where to find these for M1 and M2?

Also, thanks for merging my [PRs](https://github.com/Acurast/acurast-docs/pulls?q=is%3Apr+is%3Aclosed+author%3Atakahser) in the docs repo 👍

**Comment by pascuin** [comment]
> * I like that you included some examples into your [docs](https://docs.acurast.com/developers/introduction/). However, I identified a bug: When clicking the _Edit Code_ button and consecutively clicking the _Show Example_  button, the site crashes:

Thanks for reporting this, we have fixed it.

> * I noticed that the [acurast-processor repo](https://github.com/Acurast/acurast-processor) is currently not accessible:

As mentioned in the attached [document](https://drive.google.com/file/d/1-qNZhJGSGYs91qk2_hfscsX6MPnDQv_N/view?usp=sharing) this repository is currently private, I've invited you to it.

> * since the docs don't contain step-by-step instructions or a tutorial to guide on how to the the deliveries, I'd assume that the acurast-processor repo contains theses for all 3 milestones, is that correct? If not, could you point out where to find these for M1 and M2?

The details should be in the processor repository and in the mentioned document. Let me know if you think something is missing.
 
> Also, thanks for merging my [PRs](https://github.com/Acurast/acurast-docs/pulls?q=is%3Apr+is%3Aclosed+author%3Atakahser) in the docs repo 👍

Many thanks for opening them!

**Comment by takahser** [comment]
@pascuin FYI - I'm currently blocked by the requirement of having an Android phone to test your delivery since I only have an iPhone. However, I'm in the process to acquiring one to test your delivery. I'll update you again once I have one.

**Comment by takahser** [comment]
@pascuin FYI - I'm still blocked, but I should get my hands on an Android phone sometime next week :)

**Comment by takahser** [comment]
@pascuin - quick update, I got a physical Android phone now, but I'm struggling with setting up the Android _Wireless Debugging_ properly. Will update you when I'm making progress here.

**Comment by takahser** [comment]
@pascuin thanks for your patience and sorry for the delay here. I was finally able to run your app on a physical android phone. Since it's a Xiaomi phone, I first had to [disable MIUI optimization](https://stackoverflow.com/a/42693611) in order to make it work, otherwise, all I'd get were errors.
I tried following your [instructions](https://github.com/Acurast/acurast-processor) but it's not clear to me, how I'm supposed to start a node. Could you add them or guide me where I can find them, if they already exist?

**Comment by pascuin** [comment]
@takahser For my understanding with node you mean a node of the Acurast chain and not the Acurast Processor?

**Comment by takahser** [comment]
@pascuin the Acurast Processor is the mobile app, right?

> you mean a node of the Acurast chain

Yes exactly, I'm referring to a Substrate-based node.

**Comment by RomarQ** [comment]
Hi @takahser 

I have added you to the demo repository: https://github.com/Acurast/demo-1

Let me know if you have any question.

**Comment by takahser** [comment]
@RomarQ thanks, I checked out the repo and followed the instructions in the README.

I was able to:

- install the apk on my android phone using wireless debugging and adp
  
  ```bash
  cd ./platform-tools # downloaded from https://developer.android.com/studio/releases/platform-tools
  ./adb pair 192.168.0.20:33713 # note that the port for pairing is different from the connection port (see below)
  ./adb connect 192.168.0.20:38331
  ./adb devices
  ./adb -s 192.168.0.20:38331 install ../demo-debug-unsigned.apk
  ```

- start the nodes

  ```bash
  make start
  
  (...)
  ┌────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
  │ Acurast Testnet ⚙ Added Boot Nodes                                                                                     │
  ├────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ /ip4/127.0.0.1/tcp/54455/ws/p2p/12D3KooWLy55ZJsFcVg4PL1g7YstYQmGQ4PH2sAQJA7HDRnd46Th                                   │
  └────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
  ┌───────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
  │                                                       Network launched 🚀🚀                                                       │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Namespace                    │ zombie-ce7a399ad4f9d58f78e9c60fd1957115                                                            │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Provider                     │ native                                                                                             │
  ├──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │                                                         Node Information                                                          │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Name                         │ alice                                                                                              │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Direct Link                  │ https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:9944#/explorer                                    │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Prometheus Link              │ http://127.0.0.1:54442/metrics                                                                     │
  ├──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │                                                         Node Information                                                          │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Name                         │ bob                                                                                                │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Direct Link                  │ https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:54444#/explorer                                   │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Prometheus Link              │ http://127.0.0.1:54446/metrics                                                                     │
  ├──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │                                                         Node Information                                                          │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Name                         │ charlie                                                                                            │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Direct Link                  │ https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:54448#/explorer                                   │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Prometheus Link              │ http://127.0.0.1:54450/metrics                                                                     │
  ├──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │                                                         Node Information                                                          │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Name                         │ dave                                                                                               │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Direct Link                  │ https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:54452#/explorer                                   │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Prometheus Link              │ http://127.0.0.1:54454/metrics                                                                     │
  ├──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │                                                         Node Information                                                          │
  ├──────────────────────────────┬────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Name                         │ acurast-collator                                                                                   │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Direct Link                  │ https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:8802#/explorer                                    │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Prometheus Link              │ http://127.0.0.1:54456/metrics                                                                     │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ Parachain ID                 │ 2001                                                                                               │
  ├──────────────────────────────┼────────────────────────────────────────────────────────────────────────────────────────────────────┤
  │ ChainSpec Path               │ /var/folders/9y/ny58jn152dzcqf036v5cb4nh0000gn/T/zombie-ce7a399ad4f9d58f78e9c60fd1957115_-86712-x… │
  └──────────────────────────────┴────────────────────────────────────────────────────────────────────────────────────────────────────┘
  ```

Unfortunately, I'm not able to access the nodes through polkadotjs apps though:

image

Any idea why that happens?

**Comment by RomarQ** [comment]
Hi @takahser,

You want to access the parachain explorer, which is available at https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:8802#/explorer

For mirroring the android device I use [scrcpy](https://github.com/Genymobile/scrcpy).

**Comment by takahser** [comment]
@RomarQ 

> You want to access the parachain explorer, which is available at https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:8802#/explorer

Yes, iirc I tried that as well and got the same result. But let me try again, to be sure. I'll keep you posted.

**Comment by takahser** [comment]
@RomarQ sorry for the delay here. I re-tested and I confirm I can't connect to the node on https://polkadot.js.org/apps/?rpc=ws://127.0.0.1:8802#/explorer
Shell we have another call and check together whether we can resolve this issue?

**Comment by RomarQ** [comment]
Hi @takahser,

Yes. I will send you an email to schedule the call.

**Comment by takahser** [comment]
Just adding a short update here for transparency:
- we had various calls last and this week, the grantee was very helpful in troubleshooting what appears to be environment/machine-based issues
- in the grantee's test environment I was able to test the whole flow, using my own physical Android device
- when running the node locally on my machine, so far, I haven't been able to access it using my Android phone; according to the grantee it should work and they successfully tested it on a similar machine. I'm still trying to figure out what the issue is, although it's not the first priority, since all use cases have been smoke-tested successfully already.
- currently, a few deliverables are missing in the delivery report and the grantee is currently taking a look at it
- I wasn't able to find the test instructions for [acurast-core](https://github.com/Acurast/acurast-core) (here I successfully used `cargo test` but this information should be provided, e.g. in the README) and [acurast-processor](https://github.com/Acurast/acurast-processor) (I'm referring to automated tests here)

**Comment by takahser** [comment]
@pascuin I accepted M1, feel free to have a look at my [evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/zaturn_1_takahser.md).
I'm still working on the M2 evaluation. For M3 I'm waiting for your feedback, as discussed in the private channel.

**Comment by takahser** [comment]
@pascuin I added the [M2 evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/zaturn_2_takahser.md). There are a couple of points to improve there, feel free to have a look.

**Comment by takahser** [comment]
@pascuin thanks for the updates. I finished the [evaluation for M3](https://github.com/w3f/Grant-Milestone-Delivery/blob/3ce2cb8d72d4cb4811eb8669182f121f712e94cd/evaluations/zaturn_3_takahser.md), feel free to have a look.

**Comment by pascuin** [comment]
@takahser thanks for the evaluation we will take a look at it and get back to you as soon as they addressed.

**Review by takahser — CHANGES_REQUESTED** [review]
@pascuin thanks for updating the deliverables. I have some comments.

I ran the new tests, and it looks much better w.r.t. coverage, but I have a few comments here:
- As you already stated via DM, the _Lockdown.kt_ and _Package.kt_ classes won't be covered, since manual interaction is required. 
- Afaik you didn't comment on the _V8Executor.kt_ and _App.kt_ classes, so I wanted to check whether you plan to add tests there or if there is a reason not to do that.

Also, one test failed, even after running it multiple times:

- test report overview:

    image

- Failed "Job Fulfillment" error message:

  ```bash
  java.lang.AssertionError: Verification failed: call 1 of 1: Companion(object Companion).notify(eq(Success), any())) was not called
	  at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:63)
	  at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	  at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	  at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:64)
	  at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	  at io.mockk.MockKDsl.internalVerify(API.kt:119)
	  at io.mockk.MockKKt.verify(MockK.kt:149)
	  at io.mockk.MockKKt.verify$default(MockK.kt:140)
	  at com.acurast.attested.executor.AcurastTest.Job fulfillment(AcurastTest.kt:152)
	  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	  at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	  at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	  at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	  at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	  at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	  at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	  at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	  at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	  at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	  at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	  at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	  at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	  at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	  at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	  at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	  at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	  at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	  at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	  at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.runTestClass(JUnitTestClassExecutor.java:110)
	  at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:58)
	  at org.gradle.api.internal.tasks.testing.junit.JUnitTestClassExecutor.execute(JUnitTestClassExecutor.java:38)
	  at org.gradle.api.internal.tasks.testing.junit.AbstractJUnitTestClassProcessor.processTestClass(AbstractJUnitTestClassProcessor.java:62)
	  at org.gradle.api.internal.tasks.testing.SuiteTestClassProcessor.processTestClass(SuiteTestClassProcessor.java:51)
	  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	  at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	  at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	  at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	  at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:36)
	  at org.gradle.internal.dispatch.ReflectionDispatch.dispatch(ReflectionDispatch.java:24)
	  at org.gradle.internal.dispatch.ContextClassLoaderDispatch.dispatch(ContextClassLoaderDispatch.java:33)
	  at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:94)
	  at com.sun.proxy.$Proxy5.processTestClass(Unknown Source)
	  at org.gradle.api.internal.tasks.testing.worker.TestWorker$2.run(TestWorker.java:176)
	  at org.gradle.api.internal.tasks.testing.worker.TestWorker.executeAndMaintainThreadName(TestWorker.java:129)
	  at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:100)
	  at org.gradle.api.internal.tasks.testing.worker.TestWorker.execute(TestWorker.java:60)
	  at org.gradle.process.internal.worker.child.ActionExecutionWorker.execute(ActionExecutionWorker.java:56)
	  at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:133)
	  at org.gradle.process.internal.worker.child.SystemApplicationClassLoaderWorker.call(SystemApplicationClassLoaderWorker.java:71)
	  at worker.org.gradle.process.internal.worker.GradleWorkerMain.run(GradleWorkerMain.java:69)
	  at worker.org.gradle.process.internal.worker.GradleWorkerMain.main(GradleWorkerMain.java:74)
  ```

**Comment by pascuin** [comment]
Hey @takahser appreciate the feedback

> * Afaik you didn't comment on the _V8Executor.kt_ and _App.kt_ classes, so I wanted to check whether you plan to add tests there or if there is a reason not to do that.

We have added tests for the [_V8Executor.kt_ ](https://github.com/Acurast/acurast-processor/commit/05ede8f5f47bf0e61339468deafc96db75bf981b) as for the [_App.kt_](https://github.com/Acurast/acurast-processor/blob/main/app/src/main/java/com/acurast/attested/executor/App.kt), this is  what bootstraps the android application. Every android connected test calls that file. The readme was also update with instructions to run the android connected tests.

> Also, one test failed, even after running it multiple times:

Are you still seeing this issues with the test, we have run this on the [GitHub pipeline](https://github.com/Acurast/acurast-processor/actions/runs/4108082969/jobs/7088347979), our internal GitLab pipelines and locally without any issues.

**Comment by takahser** [comment]
@pascuin for me both the build and tests fail now, could you double-check on your side?

**Build**

```bash
acurast-processor$ ./gradlew assemble 

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/home/seraya/repos/acurast-processor/local.properties'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 403ms
```

**Tests**

```bash
acurast-processor$ ./gradlew test

FAILURE: Build failed with an exception.

* What went wrong:
A problem occurred configuring project ':app'.
> SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/home/seraya/repos/acurast-processor/local.properties'.

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 3s
```

**Comment by RomarQ** [comment]
Hi @takahser,

The issue is described in the error message:
```
SDK location not found. Define a valid SDK location with an ANDROID_HOME environment variable or by setting the sdk.dir path in your project's local properties file at '/home/seraya/repos/acurast-processor/local.properties'.
```
 You will need to create a file `/home/seraya/repos/acurast-processor/local.properties`  with the following content `sdk.dir=<ANDROID_SDK_LOCATION>`.
 
This file is usually created by the android studio.

**Comment by takahser** [comment]
@RomarQ @pascuin apologies, my bad. I accidentally ran the tests on the wrong machine, hence the error 😄
Updated test report overview:

image

It's still [the same test failing with the same error](https://github.com/w3f/Grant-Milestone-Delivery/pull/606#pullrequestreview-1280848011).
I checked the [GitHub pipeline](https://github.com/Acurast/acurast-processor/actions/runs/4108082969/jobs/7088347979) but fore some reason I was unable to find the _Job fulfillment_ test. Can you assist me in either fixing the problem or finding the test in the excerpt in the pipeline?

**Comment by RomarQ** [comment]
Hi @takahser,

I have updated the CI pipeline to produce artifacts with the test results. Please have a look [Build 549774320 on the main branch](https://github.com/Acurast/acurast-processor/suites/10890464961/artifacts/549774320)

**Comment by takahser** [comment]
@RomarQ thanks, I downloaded the artifacts and opened it in my browser, it seems to work:

image

However, when opening the [build](https://github.com/Acurast/acurast-processor/actions/runs/4138465290/jobs/7154906005) at the height of the most current commit 28d0b7aeb57e1e0963d50520c598b425ea30e607 on the _main_ branch it shows me only 20 tests (not 22):

image

Any idea where the difference comes from?

**Comment by RomarQ** [comment]
The message `89 actionable tasks: 20 executed, 69 up-to-date` means that 20 gradle task got executed, the tests got executed by one of these tasks.

**Comment by takahser** [comment]
@RomarQ (cc @pascuin) fair enough, that's also why I didn't find that test in the excerpt. 😄
I tried to run it again on my machine, that one test is still failing, but since they pass (which I verified when downloading the test artifacts from the ci build, [as indicated earlier](https://github.com/w3f/Grant-Milestone-Delivery/pull/606#issuecomment-1427367420)) I'm happy to accept it, you can find my [final M2 evaluation](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/zaturn_2_takahser.md) here. On another note, I saw that the [processor repository](https://github.com/Acurast/acurast-processor) is still private, so I'd like to ask you to make it public.
Also, I have [accepted M3](https://github.com/w3f/Grant-Milestone-Delivery/blob/master/evaluations/zaturn_2_takahser.md).

After [said repository](https://github.com/Acurast/acurast-processor) has been made public I'm going to merge this PR and internally forward the invoices for all 3 milestones. 👍

**Comment by takahser** [comment]
FYI - I removed the _to close_ label because the grantee reached out to me, telling me that they'll open source an adjusted but fully-functional version of their Android app once all deliveries are approved. Hence, I'll need to re-test this part of the delivery.

**Review by takahser — CHANGES_REQUESTED** [review]
@pascuin I've successfully built and installed the [updated processor app](https://github.com/Acurast/data-processor).
However, the UI seems to have changed entirely and I didn't find any way to connect to the node:





Could you please either update the app or provide me with detailed instructions, how to connect to the testnet?

**Comment by RomarQ** [comment]
If you find it necessary, we can do a call next week to test the workflow on your device.

**Comment by takahser** [comment]
@RomarQ ok, I'll shoot you an invite for tomorrow.

**Comment by RomarQ** [comment]
Hi @takahser,

As discussed in today's call, here is the [documentation PR](https://github.com/Acurast/acurast-docs/pull/6). It documents the setup from the example shown in the call.

The small UI improvement has also been committed to https://github.com/Acurast/data-processor.

**Comment by takahser** [comment]
@RomarQ thanks, I commented directly on the docs PR.
I've built and tested the new version of the app, looks better now. 👍
Although changing the button bg color would probably make it clearer but I'll leave it to you.

Also, sharing this for transparency here: @RomarQ and I had a call today, and we successfully tested the whole flow on my Android phone, as documented on the [PR](https://github.com/Acurast/acurast-docs/pull/6) mentioned above.

Finally, to complete the grant, the following has to be done:

- make sure the instructions in the PR are clear and complete (including the screenshots)
- make the [data-processor](https://github.com/Acurast/data-processor) repo public

**Comment by takahser** [comment]
@RomarQ sorry for not looking closer earlier, I'm fine with the docs now 😄
So the only piece that is missing here is to make the [data-processor](https://github.com/Acurast/data-processor) repo public. Lmk once that's done, and we can complete the grant.
Cc @pascuin

File: deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md
Comment (takahser) [inline deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md:35]:
```suggestion
| 8.     | Kotlin Identity                                 | https://github.com/Acurast/data-processor/blob/main/app/src/main/java/com/acurast/attested/executor/utils/Acurast.kt | Module used to display the identity of the data transmitter.                                                                                                                                                               |
```

File: deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md
Comment (takahser) [inline deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md:36]:
```suggestion
| 9.     | Kotlin IPFS                                     | https://github.com/Acurast/data-processor/blob/main/app/src/main/java/com/acurast/attested/executor/utils/IPFS.kt    | Module used to fetch the scheduled script from IPFS                                                                                                                                                                        |
```

File: deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md
Comment (takahser) [inline deliveries/Zaturn_generic-attestable-oracle-parachain_milestone1-3.md:45]:
```suggestion
| 0d.    | Testing Guide                   | https://github.com/Acurast/data-processor                                              | The documentation contains step-by-step guides for testing                                            |
```

**Review by takahser — CHANGES_REQUESTED** [review]
@pascuin @RomarQ 
I updated the links for the processor mobile app.

**Review by takahser — APPROVED** [review]

**Pull request merged**
